### PR TITLE
fix: support archive.none for OSS directory artifacts

### PR DESCRIPTION
**Pull request by jibuji**
to fix problem: #6293 
I have used the following workflow to test, and the result is correct.
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  name: test-archive-none
spec:
  entrypoint: oss-getter-and-uploader
  templates:
  - name: oss-getter-and-uploader
    inputs:
      artifacts:
      - name: oss-none-empty-dir
        path: "/mnt/input/none-empty"
        oss:
          key: "some/none-empty"
        archive:
          none: {}
      - name: oss-empty-dir
        path: "/mnt/input/empty-dir"
        oss:
          key: "some/emptydir"
        archive:
          none: {}
      - name: oss-file
        path: "/mnt/input/file"
        oss:
          key: "some/file"
        archive:
          none: {}
      - name: oss-3level
        path: "/mnt/input/3level"
        oss:
          key: "some/3level-dir"
        archive:
          none: {}
    container:
      image: 'alpine:latest'
      command: ['/bin/sh', '-c']
      args: ["cp -rf /mnt/input /mnt/output"]
      volumeMounts:
        - name: out
          mountPath: /mnt
    volumes:
      - name: out
        emptyDir: { }
    outputs:
      artifacts:
      - name: oss-none-empty-dir
        path: "/mnt/output/none-empty"
        oss:
          key: "some2/none-empty"
        archive:
          none: {}
      - name: oss-empty-dir
        path: "/mnt/output/empty-dir"
        oss:
          key: "some2/emptydir"
        archive:
          none: {}
      - name: oss-file
        path: "/mnt/output/file"
        oss:
          key: "some2/file"
        archive:
          none: {}
      - name: oss-3level
        path: "/mnt/output/3level"
        oss:
          key: "some2/3level-dir"
        archive:
          none: {}
```

Thanks to @terrytangyuan  a lot.  He helped me to rectify some code, but I failed to commit the PR because my mistake. So, I open this one and commits again.

### CODE REVIEW

**Comment by terrytangyuan** [comment]
Original review feedback is in https://github.com/argoproj/argo-workflows/pull/6296

**Review by terrytangyuan — APPROVED** [review]
LGTM. All feedback in previous review have been addressed.

**Comment by jibuji** [comment]
Thanks. What should I do next? This is my first time to contribute.

**Comment by terrytangyuan** [comment]
We'll need to wait for one of the approvers to take another look to see if there are additional feedback before merging.

File: USERS.md
Comment (alexec) [inline USERS.md:132]: Should be alphabetical

**Review by alexec — CHANGES_REQUESTED** [review]

File: USERS.md
Comment (jibuji) [reply USERS.md]: Fixed

**Comment by sarabala1979** [comment]
@jibuji PR looks good. can you add a test?

**Review by alexec — APPROVED** [review]

**Pull request merged**
